lib/evm_database_ops.rb

```diff
@@ -92,16 +92,65 @@ def self.restore(db_opts, connect_opts = {})
         db_opts[:local_file] = session.uri_to_local_path(uri)
       end
 
-      backup = PostgresAdmin.restore(db_opts)
+      prepare_for_restore(db_opts[:local_file])
+      backup_file = without_connections { PostgresAdmin.restore(db_opts) }
+
     ensure
       session.disconnect if session
     end
 
-    uri ||= backup
+    uri ||= backup_file
     _log.info("[#{db_opts[:dbname]}] database has been restored from file: [#{uri}]")
     uri
   end
 
+  private_class_method def self.without_connections
+    params = ActiveRecord::Base.remove_connection
```

Comment:
do we want to call `ActiveRecord::Base.connection_pool.disconnect!` or somethign like that to disconnect all connections in the pool?

Reply:
Probably. Is that the better way to do this?

Reply:
@carbonin I think so.  If other threads have connections, I don't believe remove_connection will disconnect those connections.  I believe [connection_pool.disconnect!](https://github.com/rails/rails/blob/71f4758c50fca5220429365c23b014458e7341f6/activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb#L430-L438) is more what we want.